cmd/network/setup_linux.go

```diff
@@ -121,21 +196,48 @@ func main() {
 	}
 	vmSwitchCmd := exec.Command(nsenter, args...)
 
-	connFile, err := vsockConn.File()
-	if err != nil {
-		logrus.Fatal(err)
-	}
 	// pass in the vsock connection as a FD to the
 	// vm-switch process in the newely created namespace
 	vmSwitchCmd.ExtraFiles = []*os.File{connFile}
-	if err := vmSwitchCmd.Start(); err != nil {
-		logrus.Fatalf("could not start the vm-switch process: %v", err)
+	return vmSwitchCmd
+}
+
+func createVethPair(defaultNsPid, peerNsPid int, defaultNSVeth, rancherDesktopNSVeth string) error {
+	veth := &netlink.Veth{
+		LinkAttrs: netlink.LinkAttrs{
+			Name:      defaultNSVeth,
+			Namespace: netlink.NsPid(defaultNsPid),
+		},
+		PeerName:      rancherDesktopNSVeth,
+		PeerNamespace: netlink.NsPid(peerNsPid),
 	}
-	logrus.Infof("successfully started the vm-switch running with a PID: %v", vmSwitchCmd.Process.Pid)
+	if err := netlink.LinkAdd(veth); err != nil {
+		return err
+	}
+	logrus.Infof("created veth pair %s and %s", defaultNSVeth, rancherDesktopNSVeth)
+	return nil
+}
 
-	if err := vmSwitchCmd.Wait(); err != nil {
-		logrus.Errorf("vm-switch exited with error: %v", err)
+func configureVethPair(vethName, ipAddr string) error {
+	veth, err := netlink.LinkByName(vethName)
```

Comment:
Given that we create the veth in `createVethPair`, can we not just return it from there and avoid the call here? (The call is probably not expensive, but explicitly passing the link around seems to be more explicit about where it came from.)

Reply:
The call to get a link by name is very cheap and only occurs once per startup time. Furthermore, the link attributes that are included in the veth struct do not exactly include the slave's name, we still need to call the `netlink.LinkByName(vethName)` anyways.